### PR TITLE
binance .markets edits

### PR DESCRIPTION
**Pull request by frosty00**

### CODE REVIEW

File: js/binance.js [hunk 1]
```diff
@@ -1003,8 +1010,11 @@ module.exports = class binance extends Exchange {
             };
             const status = this.safeString2 (market, 'status', 'contractStatus');
             const active = (status === 'TRADING');
-            const margin = this.safeValue (market, 'isMarginTradingAllowed', future || delivery);
-            const contractSize = this.safeFloat (market, 'contractSize');
+            const margin = this.safeValue (market, 'isMarginTradingAllowed', false);
+            let contractSize = undefined;
+            if (future || delivery) {
+                contractSize = this.safeFloat (market, 'contractSize', 1);
```
Comment (kroitor) [inline js/binance.js:1016]: Should we do safeNumber here?

Comment (frosty00) [reply js/binance.js:1016]: I'm using floats for the contractSize on purpose, since on line 3260 it is rounded

File: js/binance.js [hunk 2]
```diff
@@ -1889,7 +1900,7 @@ module.exports = class binance extends Exchange {
         const uppercaseType = type.toUpperCase ();
         const validOrderTypes = this.safeValue (market['info'], 'orderTypes');
         if (!this.inArray (uppercaseType, validOrderTypes)) {
-            throw new InvalidOrder (this.id + ' ' + type + ' is not a valid order type in ' + market['type'] + ' market ' + symbol);
+            throw new InvalidOrder (this.id + ' ' + type + ' is not a valid order type in market ' + symbol);
```
Comment (mrbaguvix) [inline js/binance.js:1903]:
```suggestion
            throw new InvalidOrder (`${this.id} ${type} is not a valid order type in market ${symbol}`);
```

**Review by mrbaguvix — CHANGES_REQUESTED** [review]

File: js/binance.js (hunk 2 above)
Comment (kroitor) [reply js/binance.js:1903]: @mrbaguvix that would not be transpileable and would most likely break the build due to these details:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

**Pull request merged**